### DISCUSSION TITLE
[Off Topic] USB/IP Server for macOS

### DISCUSSION THREAD

**Discussion by beriberikix** [dorssel/usbipd-win#1093] [Q&A]
I realize that this question is not directly about `usbipd-win` but I was wondering if anyone knows of a similar project for macOS? With the introduction of USB/IP client support on [Docker](https://docs.docker.com/desktop/features/usbip/) means that it's possible to have cross-platform USB access in Docker. I wrote about it [here](https://blog.golioth.io/usb-docker-windows-macos/).

**Comment by abrinkman** [comment]
Hi Jonathan, I know it's been a year. However perhaps nice to point you to this repo: [https://github.com/abrinkman/usbipd-python](https://github.com/abrinkman/usbipd-python)

It's a rough Python based proof of concept I did when wanting to demonstrate what Copilot could do. It uses libusb to implement a very simple app that allows sharing of USB devices. Works on MacOS (requires sudo for actually running the server) and tested with USB sticks and joysticks. Still would require lots of work I assume to be production ready, however its a start :)
